src/Pinch/Internal/FoldList.hs

```diff
@@ -67,10 +66,6 @@ fromFoldable :: F.Foldable f => f a -> FoldList a
 fromFoldable l = FoldList (\k r -> F.foldl' k r l)
 {-# INLINE fromFoldable #-}
 
-null :: FoldList a -> Bool
-null (FoldList f) = f (\_ _ -> False) True
```

Comment:
Ahh, yes, because you are using `foldl`; I hadn't noticed that earlier. Any reason why not use `foldr`? `foldl` seems like a slightly odd choice at first glance.